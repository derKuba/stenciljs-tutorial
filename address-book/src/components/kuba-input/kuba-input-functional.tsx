import { h } from "@stencil/core";

export const KubaInputFunctional = ({
  componentId,
  label,
  type = "text",
  value,
  setter,
}: {
  componentId: string;
  label: string;
  type: "text" | "number";
  value: string;
  setter: Function;
}) => {

  const onInput = (event) => {
    setter({ value: event.target.value });
  };
  return (
    <div class="kuba-input">
      <label htmlFor={componentId}>{label}</label>
      <input type={type} id={componentId} value={value} onInput={onInput} />
    </div>
  );
};
