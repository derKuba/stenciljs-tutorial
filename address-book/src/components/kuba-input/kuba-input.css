:host {
  display: block;
}

.kuba-input {
  margin-bottom: 25px;
}
.kuba-input label {
  margin-right: 30px;
}
