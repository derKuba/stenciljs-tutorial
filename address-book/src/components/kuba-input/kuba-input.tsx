import { Component, Host, h, Event, EventEmitter, Prop } from "@stencil/core";

@Component({
  tag: "kuba-input",
  styleUrl: "kuba-input.css",
  shadow: false,
})
export class KubaInput {
  @Prop() componentId: string;

  @Prop() label: string;

  @Prop() inputType: "text" | "number" = "text";

  @Prop() value: string;

  @Event() inputEvent: EventEmitter;

  @Prop() required;

  @Prop() min;

  @Prop() max;

  @Prop() maxlength;

  @Prop() pattern;

  private onInput = (event) => {
    this.inputEvent.emit({ value: event.target.value });
  };

  render() {

    const attributes = {
      required: this.required,
      min: this.min,
      max: this.max,
      maxlength: this.maxlength,
      pattern: this.pattern,
    }

    return (
      <Host>
        <div class="kuba-input">
          <label htmlFor={this.componentId}>{this.label}</label>
          <input
            type={this.inputType}
            id={this.componentId}
            value={this.value}
            onInput={this.onInput}
            {...attributes}
          />
        </div>
        <slot />
      </Host>
    );
  }
}
