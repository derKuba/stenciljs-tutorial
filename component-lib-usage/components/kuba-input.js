import { createEvent, h, Host, proxyCustomElement } from '@stencil/core/internal/client';

const kubaInputCss = ":host{display:block}.kuba-input{margin-bottom:25px;display:flex;justify-content:space-between;width:400px}.kuba-input label{margin-right:30px}";

const KubaInput$1 = class extends HTMLElement {
  constructor() {
    super();
    this.__registerHost();
    this.inputEvent = createEvent(this, "inputEvent", 7);
    this.inputType = "text";
    this.onInput = (event) => {
      console.log(event.target.validity);
      this.inputEvent.emit({ value: event.target.value });
    };
  }
  render() {
    const attributes = {
      required: this.required,
      min: this.min,
      max: this.max,
      maxlength: this.maxlength,
      pattern: this.pattern,
    };
    return (h(Host, null, h("div", { class: "kuba-input" }, h("label", { htmlFor: this.componentId }, this.label), h("input", Object.assign({ type: this.inputType, id: this.componentId, value: this.value, onInput: this.onInput }, attributes))), h("slot", null)));
  }
  static get style() { return kubaInputCss; }
};

const KubaInput = /*@__PURE__*/proxyCustomElement(KubaInput$1, [4,"kuba-input",{"componentId":[1,"component-id"],"label":[1],"inputType":[1,"input-type"],"value":[1],"required":[8],"min":[8],"max":[8],"maxlength":[8],"pattern":[8]}]);

export { KubaInput };
