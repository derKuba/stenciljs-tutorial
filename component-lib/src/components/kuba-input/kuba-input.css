:host {
  display: block;
}

.kuba-input {
  margin-bottom: 25px;
  display: flex;
  justify-content: space-between;
  width: 400px;
}
.kuba-input label {
  margin-right: 30px;
}
