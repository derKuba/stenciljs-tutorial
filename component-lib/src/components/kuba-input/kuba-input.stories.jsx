export default {
  title: 'Content/KubaInput',
  parameters: {},
  argTypes: {},
};

const Template = args => `<kuba-input
componentId="first-name"
label="Vorname:"
value=""
required="true"
pattern="[a-zA-Z]+"
></kuba-input>`;
export const Input = Template.bind({});
