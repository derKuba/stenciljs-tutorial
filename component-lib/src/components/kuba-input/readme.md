# kuba-input



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description | Type                 | Default     |
| ------------- | -------------- | ----------- | -------------------- | ----------- |
| `componentId` | `component-id` |             | `string`             | `undefined` |
| `inputType`   | `input-type`   |             | `"number" \| "text"` | `"text"`    |
| `label`       | `label`        |             | `string`             | `undefined` |
| `max`         | `max`          |             | `any`                | `undefined` |
| `maxlength`   | `maxlength`    |             | `any`                | `undefined` |
| `min`         | `min`          |             | `any`                | `undefined` |
| `pattern`     | `pattern`      |             | `any`                | `undefined` |
| `required`    | `required`     |             | `any`                | `undefined` |
| `value`       | `value`        |             | `string`             | `undefined` |


## Events

| Event        | Description | Type               |
| ------------ | ----------- | ------------------ |
| `inputEvent` |             | `CustomEvent<any>` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
