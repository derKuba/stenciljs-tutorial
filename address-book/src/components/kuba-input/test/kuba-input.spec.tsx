import { newSpecPage } from "@stencil/core/testing";
import { KubaInput } from "../kuba-input";

describe("kuba-input", () => {
  it("renders with toMatchSnapshot", async () => {
    const page = await newSpecPage({
      components: [KubaInput],
      html: `<kuba-input
      componentId="last-name"
      label="Nachname:"
      onInputEvent={()=>{}}
      value={""}
    ></kuba-input>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it("renders with toMatchInlineSnapshot", async () => {
    const page = await newSpecPage({
      components: [KubaInput],
      html: `<kuba-input
      componentId="last-name"
      label="Nachname:"
      onInputEvent={()=>{}}
      value={""}
    ></kuba-input>`,
    });
    expect(page.root).toMatchInlineSnapshot(`
      <kuba-input componentid="last-name" label="Nachname:" oninputevent="{()=">
        <!---->
        <div class="kuba-input">
          <label>
            Nachname:
          </label>
          <input type="text">
        </div>
        {}} value={""} &gt;
      </kuba-input>
    `);
  });
});
