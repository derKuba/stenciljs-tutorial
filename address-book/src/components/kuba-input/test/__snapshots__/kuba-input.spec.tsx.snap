// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kuba-input renders 1`] = `
<kuba-input componentid="last-name" label="Nachname:" oninputevent="{()=">
  <mock:shadow-root>
    <div class="kuba-input">
      <label>
        Nachname:
      </label>
      <input type="text">
    </div>
  </mock:shadow-root>
  {}} value={""} &gt;
</kuba-input>
`;

exports[`kuba-input renders with toMatchSnapshot 1`] = `
<kuba-input componentid="last-name" label="Nachname:" oninputevent="{()=">
  <!---->
  <div class="kuba-input">
    <label>
      Nachname:
    </label>
    <input type="text">
  </div>
  {}} value={""} &gt;
</kuba-input>
`;
